dali/pipeline/data/tensor_list.h

```diff
@@ -80,11 +78,16 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
     auto type = other[0].type();
     auto layout = other[0].GetLayout();
 
-    vector<Dims> new_shape(other.size());
+    int dim = other[0].shape().sample_dim();
+    kernels::TensorListShape<> new_shape(other.size(), dim);
     for (size_t i = 0; i < other.size(); ++i) {
+      DALI_ENFORCE(other[i].shape().sample_dim() == dim,
```

Comment:
can you use `kernels::is_uniform` here?

Reply:
It's not applicable. `is_uniform` checks if all samples in `TensorListShape` have equal dimensions. Here, we check if all Tensors in vector have the same dim.